src/orders/src/main/java/mushop/orders/config/MonitoringConfiguration.java

```diff
@@ -0,0 +1,163 @@
+import java.util.Map;
+
+@Configuration
+public class MonitoringConfiguration {
+
+    @Autowired
+    private Environment env;
+    
+    private MonitoringClient monitoringClient = null;
+    private String compartmentId = null;
+    private String namespace = null;
+    private String region = null;
+
+    public MonitoringClient getMonitoringClient() {
+	return monitoringClient;
+    }
+
+    public String getRegion() {
+	return region;
+    }
+
+    public String getNamespace() {
+	return namespace;
+    }
+
+    public String getCompartmentId() {
+	return compartmentId;
+    }
+    
+    public MonitoringClient initConnection() throws Exception {
+
+        final String tenantId = env.getProperty("OCI_TENANT_ID");
+        final String userId = env.getProperty("OCI_USER_ID");
+        final String fingerprint = env.getProperty("OCI_FINGERPRINT");
+        final String privateKey = env.getProperty("OCI_API_KEY");
+        final String passPhrase = env.getProperty("OCI_PASSPHRASE");
+        final String monitoringEndPoint = env.getProperty("OCI_MONITORING_ENDPOINT");
+        compartmentId = env.getProperty("OCI_COMPARTMENT_ID");
+        namespace = "mushopnamespace";
+        region = env.getProperty("OCI_REGION");
+    
+        System.out.println("Started Initializing the Monitoring Service...");
+        //System.out.printf("%s,%s,%s,%s,%s,%s",tenantId,userId,fingerprint,privateKey,monitoringEndPoint,compartmentId);
+        
+        if (tenantId == null || userId == null || fingerprint == null || privateKey == null){
+                System.out.printf("Cannot send metrics to OCI Monitoring, Ensure you have set the required ENV variables tenantId, userId, fingerprint and privateKey ..");
+                return null;
+        }
+
+        AuthenticationDetailsProvider provider = null;
+                
+        if (privateKey.contains("Proc-Type: 4,ENCRYPTED")) {
+                provider = SimpleAuthenticationDetailsProvider.builder()
+                .tenantId(tenantId)
+                .userId(userId)
+                .fingerprint(fingerprint)
+                .privateKeySupplier(new StringPrivateKeySupplier(privateKey))
+                .region(Region.fromRegionId(region))
+                .passPhrase(passPhrase)
+                .build();
+        } else {
+                provider = SimpleAuthenticationDetailsProvider.builder()
+                .tenantId(tenantId)
+                .userId(userId)
+                .fingerprint(fingerprint)
+                .privateKeySupplier(new StringPrivateKeySupplier(privateKey))
+                .region(Region.fromRegionId(region))
+                .build();
+        }
+                
+        monitoringClient = new MonitoringClient(provider);
+
+        if (monitoringEndPoint != null){
+                monitoringClient.setEndpoint(monitoringEndPoint);
+        } else {
+                monitoringClient.setEndpoint("https://telemetry-ingestion.us-phoenix-1.oraclecloud.com");
```

Comment:
externalize configurations and endpoints

Reply:
Also - docs to update that when deployng to regions other than Phx.

Reply:
actually the monitoring-secret collects monitoring_endpoint
 
kubectl create secret generic monitoring-secret \
--from-literal=compartment_id="<your compartment_id>" \
--from-literal=tenant_id="<your tenancy_id>" \
--from-literal=user_id="<your user_id>" \
--from-literal=fingerprint="<your fingerprint>" \
--from-literal=passphrase="<your passphrase>" \
--from-literal=monitoring_endpoint="<your monitoring endpoint>" \
--from-literal=region="<your region>" \
--from-file=apikey=<your api_key.pem>